game/player.hh

```diff
@@ -89,6 +88,7 @@ struct PlayerItem {
 	 Provides ordering and ensures id is unique.*/
 	bool operator< (PlayerItem const& pi) const
```

Comment:
not sure this is correct now: invalid id will not be stable in their ordering, this may mess up the user.

Reply:
I'm not sure this is relevant. In practice it means invalid IDs would be bunched together (which would already happen with the current code) and if we're trying to sort anything more than at most one invalid ID, we're doing something wrong IMO.

Reply:
So yeah, turns out you were right and this was the source of the bug. I reverted this part.